src/MonoMod.UnitTest/CalliTest.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Mono.Cecil.Cil;
+using MonoMod.Core.Platforms;
+using MonoMod.Utils;
+using System;
+using Xunit;
+using Xunit.Abstractions;
+
+
+namespace MonoMod.UnitTest
+{
+    public unsafe sealed class CalliTests : TestBase
+    {
+        public CalliTests(ITestOutputHelper helper) : base(helper)
+        {
+        }
+
+        [Fact]
+        public void CalliS()
+        {
+            if (PlatformDetection.Runtime is RuntimeKind.Mono)
+            {
+                LightCalli<random_struct>();
+            }
```

Comment:
Why do we need to do something different on Mono, and what are the differences in implementation?